docs/rules-cookbook.md

```diff
@@ -0,0 +1,519 @@
+
+```
+  actions:
+    backend:
+      origin: 'https://my-backend.mydomain.com'
+```
+
+### 2.2 Balancing over multiple backends
+
+Spreads the matched requests over two servers:
+
+```
+  actions:
+    backend:
+      servers:
+        - 'https://backend-1.mydomain.com'
+        - 'https://backend-2.mydomain.com'
+```
+
+### 2.3 Multiple backends with custom queue size and connection limits:
+
+```
+  actions:
+    backend:
+      servers:
+        - server: 'https://small-backend.mydomain.com'
+          max_connections: 32
+          max_queued_connections: 16
+        - server: 'https://large-backend.mydomain.com'
+          max_connections: 2048
+          max_queued_connections: 1024
+```
+
+## 3. Rewriting
+
+### 3.1 Rewriting paths
+
+Add a prefix to the request path:
+
+```
+  actions:
+    req_path:
+      - prefix:
+          add: /myprefix
+```
+
+Replace a specific part of a path regardless of case:
+
+```
+  actions:
+    req_path:
+      - replace:
+          from_exact: '/path/to/replace'
+          ignore_case: true
+          to: '/new/path'
+```
+
+Rewrite a path based on regular expression matching:
+
+```
+  actions:
+    req_path:
+      - replace:
+          from_regex: '/some/path/(.*)'
+          to_regsub: '/\1'
+```
+
+The above actually removes a the prefix '/some/path' from the path, which can also be achieved using the `prefix` structure:
+
+```
+  actions:
+    req_path:
+      - prefix:
+          remove: /some/path
+```
+
+### 3.2 Rewriting headers
+
+When manipulating headers, _southbound_ is the direction of the incoming request, i.e. toward the origin, and _northbound_ is the direction of the response, i.e. toward the client.
+
+Setting the southbound host header:
+
+```
+  actions:
+    header_southbound:
+      - set:
+          field: 'Host'
+          value: 'backend-host-name.localdomain'
+```
+
+Adding a southbound header that the backend application requires:
+
+```
+  actions:
+    header_southbound:
+      - add:
+          field: 'Authorization'
+          value: 'My-secret-authorization-token'
+```
+
+Setting various northbound headers for access control:
+
+```
+  actions:
+    header_northbound:
+      - set:
+          field: 'Access-Control-Allow-Origin'
+          value: '*'
+      - set:
+          field: 'Access-Control-Allow-Methods'
+          value: 'GET,POST'
+      - set:
+          field: 'Access-Control-Allow-Credentials'
+          value: 'false'
+      - set:
+          field: 'Access-Control-Max-Age'
+          value: '86400'
+```
+
+Removing a header sent by the origin that should not reach the client:
+
+```
+  actions:
+    header_northbound:
+      - remove: 'X-Robots-Tag'
+```
+
+## 4. Redirecting
+
+### 4.1 Simple redirection
+
+Temporary (HTTP 307) redirect of all matching requests to a new domain:
+
+```
+  actions:
+	  redirect:
+		  type: temporary
+			domain: www.redirectdomain.com
+```
+
+Permament (HTTP 308) redirect of the same type, but redirect to HTTP specifically:
```

Comment:
`s/permament/permanent/`

Reply:
Resolved in 4fcc40dfd6d8dfe14008aa4066cb3cfc1cc39af3